src/native/corehost/fxr/sdk_resolver.cpp

```diff
@@ -353,6 +420,36 @@ bool sdk_resolver::parse_global_file(pal::string_t global_file_path)
         }
     }
 
+    const auto& paths_value = sdk->value.FindMember(_X("paths"));
+    if (paths_value != sdk->value.MemberEnd() && !paths_value->value.IsNull())
+    {
+        if (!paths_value->value.IsArray())
+        {
+            trace::warning(_X("Expected an array for 'sdk/paths' value in [%s]"), global_file_path.c_str());
+            return false;
+        }
+
+        has_custom_paths = true;
```

Comment:
Is this to make a distinction between not providing a `paths` member and it being empty? If so, I'm not sure that is all that interesting. Couldn't we just assume that if the array of paths is empty, we don't have any?

Reply:
I wanted to distinguish between a empty paths being specified and the default case - basically, if custom paths are specified, that is the source of truth. I updated the resolver error to indicate that empty paths were specified if that was the case.

I kept the `has_custom_paths` since currently, if the host determines the global.json is invalid, it falls back to default resolution instead of being a hard error. It may be worth considering a breaking change to change that (there's been confusion around it in the past and it looks like some SDK commands will actually error out even though the host doesn't), but not as part of this change.